app/controllers/paused_registration_controller.rb

```diff
@@ -139,13 +141,28 @@ def get_translated_service_name(simple_id)
     CONFIG_PROXY.get_transaction_translations(simple_id, params[:locale]).fetch(:name, nil)
   end
 
-  def get_idp_list(transaction_id)
+  def get_idp_list_for_registration(transaction_id)
     list = CONFIG_PROXY.get_available_idp_list_for_registration(transaction_id, "LEVEL_2")
     return nil if list.nil?
 
     list.idps
   end
 
+  def get_idp_list_for_sign_in
+    list = CONFIG_PROXY.get_idp_list_for_sign_in(last_rp)
+    return nil if list.nil?
+
+    list.idps
+  end
+
+  def last_idp_from_those_offering_sign_in
+    get_idp_choice(get_idp_list_for_sign_in, last_idp)
+  end
+
+  def last_idp_from_those_offering_registration
```

Comment:
```suggestion
  def last_idp_offers_registration
```
Make method name more clear

Reply:
As above